src/common/StatusArg.cpp

```diff
@@ -142,6 +145,52 @@ void StatusVector::ImplStatusVector::assign(const Exception& ex) throw()
 {
 	clear();
 	ex.stuffException(m_status_vector);
+	putStrArg(0);
+}
+
+void StatusVector::ImplStatusVector::putStrArg(unsigned startWith)
+{
+	for (ISC_STATUS* arg = m_status_vector.begin() + startWith; *arg != isc_arg_end; arg += fb_utils::nextArg(*arg))
+	{
+		if (!fb_utils::isStr(*arg))
+			continue;
+
+		const char** ptr = reinterpret_cast<const char**>(&arg[fb_utils::nextArg(*arg) - 1]);
+		unsigned pos = m_strings.length();
+		const char* oldBase = m_strings.c_str();
+
+		if (*arg == isc_arg_cstring)
+		{
+			m_strings.append(*ptr, arg[1]);
+			m_strings.append(1, '\0');
```

Comment:
@AlexPeshkoff these two lines could cause to allocations in m_strings and that may makes the code fail as the first allocation can realloc m_strings to a first place and the second one may realloc it to the first place, making setStrPointers have a wrong understand of what had happened.

Reply:
If it allocates to exactly original place - no trouble, all pointers will be correct as if there was no reallocation. But if buffer is allocated not exactly at original place but overlapping it - yes, it's really trouble. Will fix.